src/providers/hana/qgshanacolumntypethread.cpp

```diff
@@ -0,0 +1,91 @@
+/***************************************************************************
+   qgshanacolumntypethread.cpp
+   --------------------------------------
+   Date      : 31-05-2019
+   Copyright : (C) SAP SE
+   Author    : Maxim Rylov
+ ***************************************************************************/
+
+/***************************************************************************
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ ***************************************************************************/
+#include "qgshanacolumntypethread.h"
+#include "qgshanaconnection.h"
+#include "qgshanautils.h"
+#include "qgslogger.h"
+#include "qgsmessagelog.h"
+#include <mutex>
+
+QgsHanaColumnTypeThread::QgsHanaColumnTypeThread( const QString &connName, const QgsDataSourceUri &uri, bool allowGeometrylessTables, bool userTablesOnly )
+  : mConnectionName( connName )
+  , mUri( uri )
+  , mAllowGeometrylessTables( allowGeometrylessTables )
+  , mUserTablesOnly( userTablesOnly )
+  , mStopped( false )
+{
+  static std::once_flag initialized;
+  std::call_once( initialized, [ = ]( )
+  {
+    qRegisterMetaType<QgsHanaLayerProperty>( "QgsHanaLayerProperty" );
+  } );
+}
+
+void QgsHanaColumnTypeThread::stop()
+{
+  mStopped = true;
+}
+
+void QgsHanaColumnTypeThread::run()
+{
+  mStopped = false;
+
+  QgsHanaConnectionRef conn( mUri );
```

Comment:
The creation of the HANA connection pool singleton is not thread-safe, but it might happen that this thread and the main thread both call `QgsHanaConnectionPool::instance()` concurrently.

Hence, we need to make `QgsHanaConnectionPool::instance()` thread-safe. The connection pool itself seems to be thread-safe.

Reply:
Done in 97d98ca.